pkg/secrets/types.go

```diff
@@ -12,6 +12,41 @@ import (
 	"github.com/stretchr/testify/mock"
 )
 
+// DecryptorError describes the error returned by a Decryptor
+type DecryptorError struct {
+	error
+	message   string
+	retriable bool
+}
+
+// Error implements the Error interface
+func (e *DecryptorError) Error() string {
+	return e.message
+}
+
+// IsRetriable returns wether the error is retriable
+func (e *DecryptorError) IsRetriable() bool {
+	return e.retriable
+}
+
+// NewDecryptorError returns a new DecryptorError
+func NewDecryptorError(message string, retriable bool) *DecryptorError {
```

Comment:
To follow golang 1.13 best practice I think the function should take a `err error` instead of `message string`

something like
```golang
type DecryptorError struct {
   error
   retriable bool
}

func NewDecryptorError(err error, retriable bool) *DecryptorError
```

Official doc: https://golang.org/pkg/errors/
interesting [blogpost](https://blog.tomarrell.com/post/introducing_wrapcheck_linter_for_go)

Reply:
Thank you! PR updated